### ISSUE TITLE
getting error on check_displayed_words

### ISSUE THREAD

**Issue by vcfvct** [kamykn/spelunker.vim#40]
Upgraded the latest version and getting Error message after scrolling in files:
```
Error detected while processing function spelunker#check_displayed_words[10]..spelunker#words#check_display_area[9]..spelunker#words#highlight[32]..spelunker#matches#delete_matches:
line   21:
E803: ID not found: 25
```

Once triggered, this error will be there all the time and have to quit vim. 😢 

macOS, nvim 0.43, Alacritty.

Not sure it is just me or applied to others.
![vim-error](https://user-images.githubusercontent.com/2096033/80152501-c7f5a600-8589-11ea-8c18-0f878642ee9d.png)

**Comment by tsuyoshicho** [comment]
This error has occured in Vim8 
```
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Apr 20 2020 21:10:30)
MS-Windows 64 ビット GUI/コンソール 版 with OLE サポート
適用済パッチ: 1-612
```

code digging to 

https://github.com/kamykn/spelunker.vim/blob/dbfde3f3b0e64a1e9e9ecd29d6f72bffdc9c39b8/autoload/spelunker/matches.vim#L67-L90

https://github.com/kamykn/spelunker.vim/blob/dbfde3f3b0e64a1e9e9ecd29d6f72bffdc9c39b8/autoload/spelunker/matches.vim#L80-L81

nvim cod need try/catch and or 2nd arg support code refactor need?

**Comment by kamykn** [comment]
@vcfvct @tsuyoshicho 
Thank you for your bug report. 
I'm sorry, in pull request #39 broke option of spellcheck displayed words in buffer. Unit test was successful, but getting error when scrolling.

`matchdelete()` is different behavior between Vim8 and Neovim v0.4.3.
Neovim current version does not supported 2nd argument of [`matchdelete()`](https://vim-jp.org/vimdoc-en/eval.html#matchdelete()). If the second argument to matchdelete () is not specified, this function will affect all buffers. (In contrast, `matchadd()` is working in the current window.)
Some times this behavior occurs problems when same buffer between split window. Second args of `matchdelete()` is resolving this problem on spelunker.vim plugin.

This issue's bug is fixed in pull request #40, but this is not radical solution. [The best way is to patched neovim `matchdelete()` function as same as Vim8](https://github.com/neovim/neovim/issues/12110), but this pull request would be support for current nvim version. (v0.4.3)

**Comment by kamykn** [comment]
> This issue's bug is fixed in pull request #40,

Sorry, the pull request number I commented yesterday was wrong.
I have corrected my comment above. (#41 is correct.)
